packages/core/src/lib/markdown-it/plugins/markdown-it-icons.ts

```diff
@@ -1,68 +1,110 @@
-      const octiconIcon = getOcticonIcon(iconFontName);
-      // ensure octicons are valid
-      if (octiconIcon === null) {
-        return '<span aria-hidden="true"></span>';
-      }
-      return iconClass
-        ? octiconIcon.toSVG({ class: iconClass })
-        : octiconIcon.toSVG();
-    } else if (iconFontType === 'octiconlight') {
-      const octiconIcon = getOcticonIcon(iconFontName);
-      // ensure octicons are valid
-      if (octiconIcon === null) {
-        return '<span aria-hidden="true"></span>';
-      }
-      const octiconIconHtml = iconClass
-        ? octiconIcon.toSVG({ class: iconClass })
-        : octiconIcon.toSVG();
-      const $ = cheerio.load(octiconIconHtml);
-      $('svg').attr('style', 'color: #fff');
-      return $.html();
-    } else if (iconFontType.startsWith('mi')) {
-      let materialIconsClass = 'material-icons';
-      switch (iconFontType) {
-      case 'mio':
-        materialIconsClass += '-outlined';
-        break;
-      case 'mir':
-        materialIconsClass += '-round';
-        break;
-      case 'mis':
-        materialIconsClass += '-sharp';
-        break;
-      case 'mit':
-        materialIconsClass += '-two-tone';
-        break;
-      default:
-      // .material-icons generates 'Filled' style icons; hence, no suffix is needed for 'mif'.
-      }
-      // Use .align-middle by default to vertically-align the icon with its surrounding text (if any).
-      // Also, replace dashes (-) with underscores (_) to format the icon name properly.
-      return `<span aria-hidden="true" class="${materialIconsClass} align-middle">`
-        + `${iconFontName.split('-').join('_')}</span>`;
-    } // If icon is a Font Awesome icon
-    return `<span aria-hidden="true" class="${iconFontType} fa-${iconFontName}"></span>`;
-  },
+  if (iconFontType === 'glyphicon') {
+    return handleGlyphicon(iconFontName);
+  } else if (iconFontType === 'octicon') {
+    return handleOcticon(iconFontName, iconClass);
+  } else if (iconFontType === 'octiconlight') {
+    return handleOcticonLight(iconFontName, iconClass);
+  } else if (iconFontType.startsWith('mi')) {
+    return handleMaterialIcon(iconFontType, iconFontName);
+  }
+  return handleFontAwesome(iconFontType, iconFontName);
+};
+
+// function to process the icon string and get the icon html
+const processIconString = (iconStr: string) => {
+  let icon = iconStr;
+
+  if (!iconStr.startsWith(':') && !iconStr.endsWith(':')) {
+    icon = `:${iconStr}:`;
+  }
+
+  const match = icon.match(ICON_REGEXP);
+  return match ? getIconHtml(match) : null;
+};
+
+// create a markdown-it plugin to process the icon strings in the markdown
+const markdownItPlugin = markdownItRegExp(
+  ICON_REGEXP,
+  (match: string[]) => getIconHtml(match),
 );
+
+// Attach `getIconHtml` to the exported function
+markdownItPlugin.processIconString = processIconString;
+
+// export the markdown-it plugin
+module.exports = markdownItPlugin;
```

Comment:
why not use a regular object property? To make it clear markdown-it does not require having a `processIconString` property inside the plugin.

```
module.exports = {
  markdownItPlugin,
  processIconString
}
```